e2e/e2etests/test_solana_deposit_restricted_address.go

```diff
@@ -0,0 +1,21 @@
+package e2etests
+
+import (
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/require"
+
+	"github.com/zeta-chain/zetacore/e2e/runner"
+)
+
+func TestSolanaDepositRestricted(r *runner.E2ERunner, args []string) {
+	require.Len(r, args, 2)
+
+	// parse restricted address
+	receiverRestricted := ethcommon.HexToAddress(args[0])
+
+	// parse deposit amount (in lamports)
+	depositAmount := parseBigInt(r, args[1])
+
+	// execute the deposit transaction
+	r.SOLDepositAndCall(nil, receiverRestricted, depositAmount, nil)
```

Comment:
Can't we do some check on the cctx here as well?

Reply:
It's checked in `poste2e_checker.go` to make sure the balance stays `0.`

The inbound is skipped by zetaclients. What I can do is a double check and make sure no cctx is created by a query. See commit https://github.com/zeta-chain/node/pull/2795/commits/77b768afb22ad0cdd6f74d6327389125e5b2a6d9